test/e2e/security_test.go

```diff
@@ -0,0 +1,122 @@
+// Copyright 2020 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package e2e
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"testing"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	restclient "k8s.io/client-go/rest"
+
+	"github.com/vmware-tanzu/antrea/pkg/apis"
+	"github.com/vmware-tanzu/antrea/pkg/apis/clusterinformation/v1beta1"
+	"github.com/vmware-tanzu/antrea/pkg/apiserver/certificate"
+)
+
+// TestSelfSignedCert first restarts antrea-controller to regenerate self-signed cert, then checks:
+// 1. The CA bundle published in antrea-ca ConfigMap can be used to verify antrea-controller's serving cert.
+// 2. The CA bundle in Antrea APIServices are the same as the one in antrea-ca ConfigMap.
+// 3. All antrea-agents can use the CA bundle to verify antrea-controller's serving cert.
+// TODO: Add TestUserProvidedCert to verify the selfSignedCert=false case.
+func TestSelfSignedCert(t *testing.T) {
```

Comment:
I feel like this is a stupid question but I'll ask anyway... Since `selfSignedCert=true` is the default case, aren't we guaranteed that it works by all the other test we run? Is the whole point of this test to check that nothing breaks if we restart the controller and a different self-signed certificate is used?

Reply:
Updating certificate is one point, another reason is it needs **all** the other test indirectly and not straightforward to figure out the real cause when some other tests fail due to this code.
For example, the channel itself only affect networkpolicy, the CA in APIServices affect antctl and deleting namespace, more importantly rotating certificate doesn't affect anything if it doesn't work.
The dedicated e2e tests can help verify the direct effect we are expecting for almost all scenarios we want to cover.